CMakeLists.txt

```diff
@@ -52,11 +53,35 @@ if (BUILD_UNIT_TESTS AND (NOT BUILD_WRITERS OR NOT BUILD_READERS))
 endif()
 
 set(BUILD_SYSTEM_DEPS_LIST AUTO ALWAYS NEVER)
+
 set_property(CACHE BUILD_SYSTEM_DEPS PROPERTY STRINGS ${BUILD_SYSTEM_DEPS_LIST})
 if(NOT BUILD_SYSTEM_DEPS IN_LIST BUILD_SYSTEM_DEPS_LIST)
     message(FATAL_ERROR "BUILD_SYSTEM_DEPS must be one of ${BUILD_SYSTEM_DEPS_LIST}")
 endif()
 
+set_property(CACHE BUILD_SYSTEM_DEPS_STB PROPERTY STRINGS ${BUILD_SYSTEM_DEPS_LIST})
+if(NOT BUILD_SYSTEM_DEPS_STB IN_LIST BUILD_SYSTEM_DEPS_LIST)
+    message(FATAL_ERROR "BUILD_SYSTEM_DEPS_STB must be one of ${BUILD_SYSTEM_DEPS_LIST}")
+endif()
+
+if ((BUILD_READERS OR BUILD_WRITERS) AND (BUILD_EXAMPLES OR BUILD_BLACKBOX_TESTS))
```

Comment:
`(BUILD_READERS OR BUILD_WRITERS)` is always true, hence can be dropped here.

Reply:
I'm not convinced `(BUILD_READERS OR BUILD_WRITERS)` is always true, when
both BUILD_READERS and BUILD_WRITERS are off and BUILD_UNIT_TESTS is
off, then BUILD_READERS and BUILD_WRITERS will not be turned back on.
Please let me know if I missed something there though.

Reply:
It might be that having both set to OFF is not technically prohibited by cmake as configured atm. but having both OFF makes no sens, the library would then do absolutely nothing. So, if you feel the need to fix that potential mis-configuration by the user then a specific check for that case should be added so that the rest of the code can rely on at least one being ON.

Reply:
Added a check for it, dropped that part of the stb check.